deepinv/datasets/lidc_idri.py

```diff
@@ -0,0 +1,139 @@
+from typing import (
+    Any,
+    Callable,
+    NamedTuple,
+    Optional,
+)
+import os
+from pydicom import dcmread
+import torch
+import numpy as np
+import pandas as pd
+
+
+class LidcIdriSliceDataset(torch.utils.data.Dataset):
+    """Dataset for `LIDC-IDRI <https://www.cancerimagingarchive.net/collection/lidc-idri/>`_ that provides access to CT image slices.
+
+    | The Lung Image Database Consortium image collection (LIDC-IDRI) consists
+    | of diagnostic and lung cancer screening thoracic computed tomography (CT)
+    | scans with marked-up annotated lesions.
+    | It is a web-accessible international resource for development, training,
+    | and evaluation of computer-assisted diagnostic (CAD) methods for lung
+    | cancer detection and diagnosis.
+    | To download the dataset, you will need to install the `NBIA Data Retriever <https://wiki.cancerimagingarchive.net/display/NBIA/Downloading+TCIA+Images>`_.
```

Comment:
I would give more detailed instructions regarding the download and set up, and add a warning box (.. warning: ) saying that this class does not download the dataset, and it is up to the user to download the raw files.